### DISCUSSION TITLE
Feel++ Debian packages are now delivered by JFrog

### DISCUSSION THREAD

**Discussion by prudhomm** [feelpp/feelpp#1638] [Announcements]
Bintray.com has served  the Feel++ Debian packages for some time but the service was shut down beginning of May 2021.
We have move the service to https://feelpp.jfrog.io and updated our procedure 

For example for Ubuntu Focal 20.04, follow these steps to get the Feel++ core toolboxes and mor packages
```
echo "deb https://feelpp.jfrog.io/artifactory/ubuntu focal latest" | sudo tee -a /etc/apt/sources.list
wget -qO - https://feelpp.jfrog.io/artifactory/api/security/keypair/gpg-debian/public | sudo apt-key add -
apt update
```

Debian packages for focal and buster are automatically generated when new commits are done in the `develop` branch and then part of the `latest` distribution channel. 
When a release is done, the master branch is updated and Debian packages for focal and buster are generated for the `stable` distribution channel.
Of course, Debian packages can be used in Windows 10 thanks to WSL and the installation of a subsystem like focal or buster.

More details are available [here](https://docs.feelpp.org/user/0.109/install/index.html)

**Comment by Trophime** [comment]
A quick comment. On recent debian/ubuntu distribution there is no apt-key command anymore.
To properly work-around this you can do (as root) something like that:

```
wget -qO - https://feelpp.jfrog.io/artifactory/api/security/keypair/gpg-debian/public | gpg --dearmor > feelpp.gpg
install -o root -g root -m 644 feelpp.gpg /etc/apt/trusted.gpg.d
echo "deb [signed-by=/etc/apt/trusted.gpg.d/feelpp.gpg] https://feelpp.jfrog.io/artifactory/ubuntu focal ${CHANNEL}" | tee -a /etc/apt/sources.list.d/feelpp.list
        
```
where CHANNEL variable points to latest.

An another note, as for me I prefer to keep jforg repository in a specific apt source file (ie feelpp.list) instead of adding it to the main apt source file defintions.

**Reply by prudhomm** [reply]
@Trophime thanks a lot for the comment.
could you update the document of feelpp [here](https://docs.feelpp.org/user/0.109/install/index.html) please ?

**Comment by Trophime** [comment]
@prudhomm just done it by creating a PR

**Reply by Trophime** [reply]
@prudhomm I cannot get feelpp from jfrog. Seems like we have a problem here:
```
{
  "errors" : [ {
    "status" : 429,
    "message" : "You have reached your monthly quota for Data Transfer. To continue and start enjoying unlimited transfer, upgrade your subscription"
  } ]
}
```

**Reply by Trophime** [reply]
We really shall consider having our debian repository for feelpp.
Building such kind of repo with aptly does not seem so difficult.
It just requires some disk space and a "machine" (eventually a container)

**Reply by vincentchabannes** [reply]
Indeed, Jfrog seems not usable if we don't pay.

**Reply by Trophime** [reply]
Seems like we already reach the limit this month:
```
Err:5 https://feelpp.jfrog.io/artifactory/ubuntu focal InRelease                                     
  429  Too Many Requests [IP: 34.105.229.119 443]
```
RRRR

**Reply by Trophime** [reply]
I suggest giving a trial to https://hub.docker.com/r/smirart/aptly
